jsparagus/emit/rust.py

```diff
@@ -200,18 +218,33 @@ def write_condition(self, state, first_act):
             if len(state.epsilon) == 1:
                 # This is an attempt to avoid huge unending compilations.
                 _, dest = next(iter(state.epsilon), (None, None))
-                self.write("// parser.top_state() in [{}]", " | ".join(map(str, first_act.states)))
+                pattern = rust_range(extract_ranges(first_act.states))
+                self.write("// parser.top_state() in ({})", pattern)
                 self.write_epsilon_transition(dest)
             else:
                 self.write("match parser.top_state() {")
                 with indent(self):
+                    max_weight = max(len(act.states) for act, dest in state.edges())
+                    max_states = []
+                    max_dest = None
                     for act, dest in state.edges():
                         assert first_act.check_same_variable(act)
-                        self.write("{} => {{", " | ".join(map(str, act.states)))
+                        if max_dest is None and max_weight == len(act.states):
+                            max_states = act.states
+                            max_dest = dest
+                            continue
+                        pattern = rust_range(extract_ranges(act.states))
+                        self.write("{} => {{", pattern)
                         with indent(self):
                             self.write_epsilon_transition(dest)
                         self.write("}")
-                    self.write("_ => panic!(\"Unexpected state value.\")")
+                    # self.write("_ => panic!(\"Unexpected state value.\")")
```

Comment:
why is this kept and commented out?

Reply:
This is some legacy due to multiple iteration on this code.